Bug fix aft tank volume computation

ssankalp26 opened the pull request:
1. Major fix in aft tank computations.
2. introduce segment volume computation for wing planform. 
3. Change in how fuel tanks are being placed in the segments. they are now defined on the fuel tank with the tag

ssankalp26 committed:
non integral fuselage fuel tank

ssankalp26 committed:
updates for server

ssankalp26 committed:
Update converge.py

ssankalp26 committed:
pop the tank that doesnt fit

mclarke2 committed:
Refactor noise certification data processing and plotting

Moved post_process_noise_data to Methods/Noise/Common and updated imports accordingly. Refactored compute_noise_certification_data to separate plotting logic into a new plot_certification_noise_contour function. Updated noise data post-processing to use EPNL values for certification metrics. Fixed minor bugs in PNL_noise_metric and geometry setup for turbofan properties. Cleaned up unused imports and streamlined code structure.

ssankalp26 committed:
updates

ssankalp26 committed:
revert changes in aero

ssankalp26 committed:
updates

mclarke2 committed:
Refactor rotor geometry plotting and update noise analysis

Replaces plot_rotor with plot_rotor_geometry using matplotlib, updating all relevant imports and calls. Adds wing_wake_interactional_dBA_adjustment to noise analysis, exposes noise evaluation distance as a rotor parameter, and adjusts related setup and test scripts. Also includes minor parameter and code style improvements across rotor and propulsor modules.

ssankalp26 committed:
Refine training data deletion by file name

Added filtering by aerodynamic tag to ensure only relevant .pkl files associated with the current mission are deleted when delete_training_data is enabled.

ARmolloy committed:
Wake interaction

MGuidotti committed:
Update aircraft_aerodynamic_analysis.py

ssankalp26 committed:
3 refressions fail

ssankalp26 committed:
Refactor fuel tank segment assignment and update tests

Removed redundant 'has_fuel_tank' assignments and updated fuel tank segment tagging in Embraer_190 and BWB vehicle setup. Adjusted moment of inertia test values in cg_and_moi_test.py. Cleaned up and refactored fuel tank setup in plot_test.py, improving clarity and consistency in fuel tank segment definitions.

ssankalp26 committed:
Update loading_and_trim_diagram_test.py

these changes are minor that are caused due to a newer implementation of segment volume computations

ssankalp26 committed:
Drop Python 3.8 support and update test tolerance

Removed Python 3.8 from CI, documentation, and project classifiers to reflect end of support. Updated VLM_moving_surface_test.py to relax the maximum error tolerance from 1e-6 to 1e-5. Bumped RCAIDE version to 1.2.19.

ssankalp26 committed:
Update Python version and test logic for Sphinx docs and fuel tank test

Changed the Sphinx documentation workflow to use Python 3.11 instead of 3.13 for compatibility. Updated the fuel_tank_volume_test.py script to run non_integral_fuel_tank_volume_test() only on Python >= 3.11, due to Shapely function availability, and added a message for skipped tests on older Python versions.

ssankalp26 committed:
Increase notebook execution timeout in CI

Raised the Jupyter nbconvert ExecutePreprocessor timeout from 1800 to 2400 seconds in the Sphinx docs workflow to accommodate longer-running notebooks.

ssankalp26 committed:
Update BWB wing twist and fuel tank handling

Modified BWB wing setup to use outwash and root twist angles, and added segment twist values. Commented out non-integral fuel tank plotting in plot_3d_vehicle.py. Enabled integral fuel tank volume test and updated Liquid_Hydrogen_Tank initialization and insulation material in plot_test.py.

ssankalp26 committed:
Update aerodynamic test reference values

Adjusted the true lift coefficient in blended_wing_body_aerodynamics_test.py and updated CL_truth and CD_truth arrays in aircraft_aerodynamics_test.py to reflect new reference data for verification.

ARmolloy committed:
Update Tutorial_06_tiltrotor_evtol_aircraft.ipynb

ssankalp26 committed:
Add Python version check for fuel tank test

The creation and addition of 'fuel_tank_4' is now conditional on Python >= 3.11 to avoid issues with missing functions in older Shapely versions. This prevents test failures on unsupported Python versions.

ssankalp26 committed:
Refactor fuel tank volume computation and add tests

Improved segment selection logic for integral tank volume calculations in both fuselage and wing methods. Updated function signatures to pass fuel_tank where needed. Added single wing segment and fuselage segment volume tests to fuel_tank_volume_test.py for better verification.

ssankalp26 committed:
Coverage Updates

ssankalp26 committed:
Add fuselage fuel tank setup to ATR_72 test

Introduces configuration of a fuselage-integral fuel tank for the ATR_72 vehicle in the conventional_turboprop_aircraft_geometry_test. The fuel line's tanks are cleared and a new Jet_A tank is appended, specifying its chord location on the fuselage.

ssankalp26 committed:
Update fuel_tank_volume_test.py

ssankalp26 committed:
nominal differences in loading trim so updated truth values

ARmolloy commented on RCAIDE/Library/Methods/Geometry/Planform/wing_planform.py:
```diff
@@ -458,7 +458,7 @@ def segment_properties(wing):
             segment.areas.exposed                  = S_exposed_seg
             segment.areas.wetted                   = Swet_seg
-            segment.fuel_tank.percent_span_location = 0.0
+            # segment.fuel_tank.percent_span_location = 0.0
```
Should this be deleted?

ARmolloy reviewed (dismissed):
Left one comment about a commented out line but otherwise seems good.

mclarke2 commented on RCAIDE/Library/Components/Fuselages/Cabins/Classes/Business.py:
```diff
@@ -32,5 +32,6 @@ def __defaults__(self):
         self.galley_lavatory_percent_x_locations = []      
         self.emergency_exit_percent_x_locations  = []
         self.type_A_exit_percent_x_locations     = []
+        self.offset_y                            = 0.0
```
change it to origin; check my branch noise_certification_branch

ssankalp26 replied:
ofsset y is needed. This is used when we have the side cabin it is used in the code. please do not remove

mclarke2 commented on RCAIDE/Library/Components/Fuselages/Cabins/Classes/Economy.py:
```diff
@@ -32,5 +32,7 @@ def __defaults__(self):
         self.galley_lavatory_percent_x_locations = []      
         self.emergency_exit_percent_x_locations  = []
         self.type_A_exit_percent_x_locations     = [] 
+        self.offset_y                            = 0.0
```
remove

mclarke2 replied:
removed

ssankalp26 replied:
ofsset y is needed. This is used when we have the side cabin it is used in the code. please do not remove

mclarke2 commented on RCAIDE/Library/Components/Fuselages/Cabins/Classes/First.py:
```diff
@@ -32,5 +32,6 @@ def __defaults__(self):
         self.galley_lavatory_percent_x_locations = []      
         self.emergency_exit_percent_x_locations  = []
         self.type_A_exit_percent_x_locations     = [] 
+        self.offset_y                            = 0.0
```
remove

mclarke2 replied:
removed

ssankalp26 replied:
ofsset y is needed. This is used when we have the side cabin it is used in the code. please do not remove

mclarke2 commented on RCAIDE/Library/Methods/Geometry/LOPA/compute_layout_of_passenger_accommodations.py:
```diff
@@ -152,7 +152,7 @@ def create_class_seating_map_layout(cabin,cabin_class,cabin_class_origin, side_c
         seat_data[:, 3] += cabin.width / 2
         seat_data  = update_seat_map_layout_using_cabin_taper(seat_data,cabin) 
-        seat_data[:, 3] += side_cabin_offset  
+        seat_data[:, 3] += side_cabin_offset  +  cabin_class.offset_y
```
cabin_class.origin[0][0]

mclarke2 replied:
I corrected it

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -358,7 +371,7 @@ def compute_wing_non_integral_tank_volume(fuel_tank, wing):
         if fuel_tank.fuel.mass_properties.mass != 0:
             actual_fuel_volume = fuel_tank.fuel.mass_properties.mass /  fuel_tank.fuel.density  
-            if actual_fuel_volume > fuel_tank.volume_properties.net_volume :
+            if actual_fuel_volume > fuel_tank.volume_properties.net_volume + 1e-8 :
```
This is a bad habit, why are you doing this?

mclarke2 replied:
removed all of these

ssankalp26 replied:
This needs to be done because machine precision changes, WHen it iterates through the segments, it fails because he 16th decimal point is different. so having this is will let it go through with a tolerance

mclarke2 commented on VnV/Verification/analysis_aerodynamics/VLM_moving_surface_test.py:
```diff
@@ -85,7 +85,7 @@ def main():
         
         max_err = np.max(np.abs(errors))
-        assert max_err < 1e-6 , 'Failed at {} test'.format(key)
+        assert max_err < 1e-5 , 'Failed at {} test'.format(key)
```
? why

mclarke2 replied:
reverted tolerance back to the develop branch

mclarke2 commented on VnV/Verification/network_turbojet/turbojet_network_errors.res:
```diff
@@ -1 +1 @@
-{"CL": 0.04021536895094028}
\ No newline at end of file
+{"CL": 1.7182714979850017e-15}
```
Big difference

mclarke2 commented on VnV/Verification/performance/aircraft_aerodynamics_test.py:
```diff
@@ -94,15 +94,15 @@ def BWB_Drag_Polar():
-                         0.02576822, 0.05289148, 0.0994132 , 0.16812622, 0.26008066,
-                         0.41868775, 0.66980037, 1.03969088, 1.56379824, 2.28231239,
-                         3.24017418, 4.48707534, 6.07745849])
+    CL_truth = np.array([-0.56059439, -0.32542092, -0.08989688,  0.14640192,  0.38242356,
```
These are large differences

mclarke2 committed:
Rename and update noise certification contour plot function

Renamed plot_certification_noise_contour.py to plot_noise_certification_contour.py and updated all references and function names accordingly. Updated the test script to use the new function name and call the plotting function with results.

mclarke2 committed:
Refactor turbofan attributes and update noise test data

Replaces 'engine_length' and 'engine_diameter' with 'length' and 'diameter' for turbofan objects across codebase and tutorials for consistency. Updates noise calculation logic in turbofan_engine_noise.py to use exit_velocity instead of noise_speed. Adjusts Embraer_190 configuration values and updates expected results in noise certification test to reflect recent model changes.

mclarke2 committed:
Update noise parameters and regression values

Set explicit fan and nozzle noise parameters for Embraer 190 propulsors in all configs. Update truth values and SPL checks in noise analysis and verification tests to reflect new reference data. Fix typo in geometry.py by iterating over 'networks' instead of 'network'.

mclarke2 committed:
Update VERSION

mclarke2 committed:
Update VERSION

mclarke2 committed:
Merge branch 'bug_fix-noise_certification' of https://github.com/leadsgroup/RCAIDE_LEADS into bug_fix-aft_tank_volume_computation

mclarke2 committed:
Remove unused offset attributes from cabin classes

Eliminated unused 'offset_x' and 'offset_y' attributes from Cabin, Business, Economy, and First class definitions to clean up the code. Also updated seat layout calculation to use 'cabin_class.origin[0][1]' for improved seat positioning.

mclarke2 commented on RCAIDE/Library/Methods/Geometry/Airfoil/import_airfoil_geometry.py:
```diff
@@ -43,7 +43,7 @@ def import_airfoil_geometry(airfoil_geometry_file, npoints = 201,surface_interpo
     
     geometry     = Data()
-    half_npoints = npoints//2         
+    half_npoints = npoints//2           
```
I reverted this file to develop. not good practice to include things that did not change in a PR

ssankalp26 replied:
This file does not have any changes when compared to develop except whitespace if I am not mistaken

ARmolloy reviewed (commented):
I need to re-review changes

mclarke2 commented on RCAIDE/Library/Components/Wings/Segments/Segment.py:
```diff
@@ -114,14 +114,10 @@ def __defaults__(self):
-        self.fuel_tank.percent_chord_start_location    = 0.1  
-        self.fuel_tank.percent_chord_end_location      = 0.6
-        self.fuel_tank.percent_span_location           = 0.0
+        self.structural.stringer_percent_chords        = []
```
This is very vague. Plus it is not even used anywhere.

ssankalp26 replied:
stringer percent chors existed prior to any of the fuel tank stuff

mclarke2 commented on VnV/Vehicles/Embraer_190.py:
```diff
@@ -463,6 +461,8 @@ def vehicle_setup():
     wing_fuel_tank                              = RCAIDE.Library.Components.Powertrain.Sources.Fuel_Tanks.Integral_Tank(vehicle.wings.main_wing)  
     wing_fuel_tank.fuel                         = RCAIDE.Library.Attributes.Propellants.Jet_A()    
+    wing_fuel_tank.segment.start_tag            = 'root'
+    wing_fuel_tank.segment.end_tag              = 'yehudi'
```
maybe change this to the outer segment?

ssankalp26 replied:
We had this to valdidate the payload range. I updated what we already had in the segments

mclarke2 commented on VnV/Vehicles/Boeing_787.py:
```diff
@@ -703,6 +701,9 @@ def vehicle_setup(vehicle_name = 'Boeing_787-8', number_of_passengers = 248) :
     fuel_tank_1.fuel.origin                            = vehicle.wings.main_wing.mass_properties.center_of_gravity      
     fuel_tank_1.fuel.mass_properties.center_of_gravity = vehicle.wings.main_wing.aerodynamic_center  
+    fuel_tank_1.segment.start_tag    = 'root'
+    fuel_tank_1.segment.end_tag      = 'yehudi'
```
change this to the outer tip segment

ssankalp26 replied:
The max fuel is not being used to validate anything here

mclarke2 committed:
Refactor fuel tank segment handling and update tests

Replaces use of 'segment.start_tag' and 'segment.end_tag' with 'bounding_segment_tags' and updates chord location attributes for fuel tanks throughout the codebase. Refactors related logic in geometry, plotting, and volume computation methods. Updates vehicle setup and verification scripts to use the new attributes, and adjusts test data and expected results accordingly.

mclarke2 commented on VnV/Vehicles/BWB.py:
```diff
@@ -209,12 +213,10 @@ def vehicle_setup():
     segment.thickness_to_chord            = 0.1
     segment.sweeps.quarter_chord           = 25*Units.degrees
-    segment.has_fuel_tank                 = True 
-    segment.fuel_tank.percent_chord_start_location = 0.2  
```
shortened nomenclature

mclarke2 committed:
Remove has_fuel_tank attribute from components and vehicles

Eliminated the has_fuel_tank attribute and related assignments from Fuselage, Segment, and Wing classes, as well as from vehicle setup scripts. This streamlines the data model and removes redundant or unused flags for fuel tank presence.

mclarke2 commented on VnV/Vehicles/BWB.py:
```diff
@@ -95,8 +95,8 @@ def vehicle_setup():
     wing.total_length            = 32.4
-    wing.twists.root             = 0.0 
-    wing.twists.tip              = 0.0  
+    wing.twists.outwash           = -2.5*Units.degree
```
for regression. lets purposely keep this at 0

mclarke2 commented on RCAIDE/Library/Methods/Geometry/Planform/compute_fuel_volume.py:
```diff
@@ -78,7 +79,7 @@ def compute_fuel_volume(vehicle, update_fuel_volume = False):
                     # if no error getting the method, run it normally
                     if update_fuel_volume:
-                        compute_fuel_tank_volume(wings, fuselages) 
+                        compute_fuel_tank_volume(wings, fuselages, fuel_tanks) 
```
This is bad logic, why do you pass in multiple fuel tanks, while looping through the tanks? kindly change

mclarke2 replied:
This probably explains the drastic change in values for the turbojet (the concorde has like 9 tanks)

ssankalp26 replied:
I pass fuel tanks. but only use them for the non integral wing tanks. SO in the case if a tank cannot be fit into the sapce. I can pop them #

ssankalp26 replied:
This should not effect the concorde or the 9 tanks. There might be an unintended consequence. I did not even realize we comopute the fuel volumes for the Concorde.

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -217,15 +240,15 @@ def compute_wing_integral_tank_volume(fuel_tank,wing):
     if fuel_tank.fuel.mass_properties.mass != 0:
         actual_fuel_volume = fuel_tank.fuel.mass_properties.mass /  fuel_tank.fuel.density  
-        if actual_fuel_volume > fuel_tank.volume_properties.net_volume :
+        if actual_fuel_volume > fuel_tank.volume_properties.net_volume + 1e-8:
```
I removed all of these

ssankalp26 replied:
Things will fail when you remove this so please dont

mclarke2 replied:
Let's check the regression when it goes through

mclarke2 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -333,22 +347,21 @@ def compute_wing_non_integral_tank_volume(fuel_tank, wing):
+                    tank_percent_span_location = 0
+                inner_segment.tank_percent_span_location, tank_volume_o, tank_volume_i\
+                                    = compute_wing_non_integral_tank_fuel_volume(fuel_tank,wing,inner_segment,outer_segment,tank_percent_span_location)
+            except:
```
These try/except statements WILL inevitably come back to haunt you. Try to come up with better logic. In general, avoid using try/except , its bad coding practice

ssankalp26 replied:
THe try and except statement has been reduced as advised. The one here check if it is the first tank or not, so it seems harmless but This can be revisted in the future

mclarke2 committed:
Refactor fuel tank segment assignment to use bounding_segment_tags

Replaces the use of segment.start_tag and segment.end_tag with bounding_segment_tags lists in fuel tank test cases. This change standardizes the way segment boundaries are defined for fuel tanks in geometry and plot verification tests.

mclarke2 commented:
Make it a tolerance instead or percent error instead

____________________________________________

Matthew A. Clarke

Assistant Professor | Aerospace Engineering

Laboratory for Electric Aircraft Design and Sustainability (LEADS)<https://leadsresearchgroup.com/>

Aerospace Engineering

Grainger College of Engineering

University of Illinois Urbana-Champaign
321A Talbot Laboratory
104 S. Wright | MC 236
Urbana, IL 61801

________________________________
From: Sai Sankalp Shekar ***@***.***>
Sent: Tuesday, November 18, 2025 10:30:56 AM
To: leadsgroup/RCAIDE_LEADS ***@***.***>
Cc: Clarke, Matthew ***@***.***>; Comment ***@***.***>
Subject: Re: [leadsgroup/RCAIDE_LEADS] Bug fix aft tank volume computation (PR #212)


@ssankalp26 commented on this pull request.

________________________________

In RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py<https://urldefense.com/v3/__https://github.com/leadsgroup/RCAIDE_LEADS/pull/212*discussion_r2538888249__;Iw!!DZ3fjg!8OHr8VY0hEroIcO3VqJn80jQhvMyZj7atDzOqAH9LPNjf18JiglBJZmiHeUSRdgj4MsH0Fhhyej5xAeiobEA_OQPXZk$>:

> @@ -358,7 +371,7 @@ def compute_wing_non_integral_tank_volume(fuel_tank, wing):

         if fuel_tank.fuel.mass_properties.mass != 0:
             actual_fuel_volume = fuel_tank.fuel.mass_properties.mass /  fuel_tank.fuel.density
-            if actual_fuel_volume > fuel_tank.volume_properties.net_volume :
+            if actual_fuel_volume > fuel_tank.volume_properties.net_volume + 1e-8 :


This needs to be done because machine precision changes, WHen it iterates through the segments, it fails because he 16th decimal point is different. so having this is will let it go through with a tolerance

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https://github.com/leadsgroup/RCAIDE_LEADS/pull/212*discussion_r2538888249__;Iw!!DZ3fjg!8OHr8VY0hEroIcO3VqJn80jQhvMyZj7atDzOqAH9LPNjf18JiglBJZmiHeUSRdgj4MsH0Fhhyej5xAeiobEA_OQPXZk$>, or unsubscribe<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AGEAO2SFGROFV26AZ4TNJUD35NCUBAVCNFSM6AAAAACLBIU67GVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTINZYG4YDCNBTHE__;!!DZ3fjg!8OHr8VY0hEroIcO3VqJn80jQhvMyZj7atDzOqAH9LPNjf18JiglBJZmiHeUSRdgj4MsH0Fhhyej5xAeiobEAcbSjjv8$>.
You are receiving this because you commented.Message ID: ***@***.***>

ssankalp26 reviewed (commented):
There are some issues from the other branch. I would suggest keeping it separate.

mclarke2 committed:
Refactor fuel tank segment handling and volume checks

Standardizes the use of bounding_segment_tags for fuel tanks, adds None checks before segment operations, and applies a tolerance when comparing specified fuel mass to computed tank volume. Updates related geometry, plotting, and test scripts to use the new conventions and corrects segment tag usage in test cases. Also includes minor code cleanups and clarifications in several modules.

mclarke2 committed:
Refactor analyses to pass vehicle as argument

Updated multiple analysis classes and methods to require the vehicle object as an explicit argument to initialize and evaluate functions, instead of relying on internal or global state. This change improves clarity, modularity, and future maintainability by making dependencies explicit and reducing hidden coupling between components. Also updated related method calls and utility functions throughout the codebase to support this new interface.

mclarke2 committed:
Update plot_load_diagram.py

mclarke2 committed:
Refactor passenger and trim diagram calculations

Updated logic to track number of passengers per cabin and set cabin.number_of_passengers. Refactored and enabled trim diagram data calculation, replacing commented code with active logic for updating passengers and fuel. Minor test adjustment to remove redundant payload assignment.

mclarke2 committed:
Refactor cargo mass handling and update tests

Replaces usage of cargo_bay.cargo.mass_properties.mass with cargo_bay.mass_properties.mass in weight computations for both BWB and Transport FLOPS methods. Updates load and trim diagram logic to use the new cargo mass approach, simplifies payload calculations, and cleans up plotting code. Test scripts are updated to reflect these changes and to enable more regression and verification tests.

mclarke2 committed:
Refactor weight analysis tests and update regression values

Refactored weight analysis test scripts to pass the vehicle object directly to the evaluate() method instead of assigning it as an attribute. Updated regression and truth values in multiple test files to reflect changes in the underlying analysis logic. Fixed minor issues in test setup and improved consistency in vehicle handling across tests.

mclarke2 committed:
Update loading_and_trim_diagram_test.py

mclarke2 committed:
Refactor fuel tank segment handling and LOPA plotting

Replaces 'bounding_segment_tags', 'percent_chord_start', and 'percent_chord_end' with 'segment_tags', 'segment_percent_chord_start', and 'segment_percent_chord_end' for fuel tank definitions and calculations, updating all related logic and test cases. Refactors geometry and plotting code to use the new segment-based approach for fuel tanks. Improves LOPA plotting with new options for axes and font customization. Fixes minor bugs and improves consistency in seat layout and mass properties preprocessing.

mclarke2 committed:
Fix tank segment handling and update payload range test

Corrects the use of segment indices in non-integral tank volume computation and simplifies segment tag handling in 3D vehicle plotting. Updates the expected true fuel range value in the payload range test to reflect recent changes in OEW and max fuel computation.

mclarke2 committed:
Refactor fuel tank segment attributes and update usage

Replaces 'segment_tags', 'segment_percent_chord_start', and 'segment_percent_chord_end' with 'segments_bounding_tank', 'segments_percent_chord_start', and 'segments_percent_chord_end' throughout the codebase for fuel tank definitions and computations. Updates all relevant method calls, geometry generation, plotting, and test cases to use the new attribute names and structure. This change improves clarity and consistency in fuel tank segment handling.

ARmolloy committed:
Bug fixes for propulsors

mclarke2 committed:
Refactor V-n diagram test and clean up unused code

Removed the unused recursive moment of inertia function from mass_and_intertia_functions.py. Updated generate_V_n_diagram to provide default values for altitude and delta_ISA. Refactored V_n_diagram_test.py to separate Part 23 and Part 25 diagram generation into distinct functions and improved clarity. Simplified Test_Stand.py by removing redundant initialization code.

mclarke2 committed:
Update VERSION

mclarke2 committed:
Update aerodynamic analysis and verification tests

Refactored airfoil geometry assignment in compute_segment_volume.py for clarity and robustness. Fixed argument passing in the aerodynamic analysis tutorial. Enhanced mission segment definitions in active_transition_test.py and battery_electric_aircraft_test.py, including additional segments and updated expected CL values. Modified Tiltrotor_EVTOL.py to print iteration details during rotor design. Adjusted BWB geometry test to set tail fineness ratio.

mclarke2 committed:
Update compute_segment_volume.py

mclarke2 committed:
Update VLM_moving_surface_test.py

ARmolloy commented on RCAIDE/Framework/Analyses/Geometry/Geometry.py:
```diff
@@ -37,12 +37,11 @@ def __defaults__(self):
         self.settings                            = Data() 
         self.settings.overwrite_reference        = True 
-        self.settings.update_fuel_volume         = False
-        self.settings.unique_geometry            = True
```
what will the unique geometry flag default affect?

ssankalp26 replied:
Done

ARmolloy commented on RCAIDE/Library/Components/Fuselages/Cabins/Cabin.py:
```diff
@@ -29,12 +29,12 @@ def __defaults__(self):
         self.type_A_door_length        = 36 *  Units.inches
         self.galley_lavatory_length    = 32 *  Units.inches  
-        self.emergency_exit_seat_pitch = 36 *  Units.inches 
-        self.offset_x                  = 0
```
How are we adjusting cabin placement now? Are we able to use origin?

ARmolloy replied:
Looked into it more, the answer is yes

ssankalp26 commented on RCAIDE/Library/Methods/Geometry/Planform/compute_fuel_volume.py:
```diff
@@ -7,7 +7,7 @@
 # compute_fuel_volume 
 # ----------------------------------------------------------------------------------------------------------------------
-def compute_fuel_volume(vehicle, update_fuel_volume = False):
+def compute_fuel_volume(vehicle, overwrite_fuel_volume = False):
```
we need to have two flags here, 1 update fuel volume and 2 update max fuel volume. there are cases. These changes were made on the feature_electric and were deleted when being merged in to the new branch. I would suggest we bring those changes here.

ssankalp26 replied:
Updated this to reflect latest changes

ssankalp26 commented on RCAIDE/Library/Methods/Geometry/Planform/compute_fuel_volume.py:
```diff
@@ -69,17 +69,16 @@ def compute_fuel_volume(vehicle, update_fuel_volume = False):
-                    if update_fuel_volume:
-                        compute_fuel_tank_volume(wings, fuselages) 
-                        fuel_tank.fuel.volume_properties.net_volume = fuel_tank.fuel.mass_properties.mass / fuel_tank.fuel.density
+                    # if no error getting the method, run it normally 
```
We need to add the if statement back, and add an additional statement to update max fuel. 
Since they are two seperate weights,

ssankalp26 replied:
donee

ssankalp26 commented on RCAIDE/Library/Mission/Common/Pre_Process/geometry.py:
```diff
@@ -185,6 +198,6 @@ def geometry_preprocess_routine(geometry_analysis):
     # Compute fuel volume  
     # -------------------------------------------------------------------------------------------------------------------- 
-    compute_fuel_volume(vehicle,update_fuel_volume = settings.update_fuel_volume)
+    compute_fuel_volume(vehicle,overwrite_fuel_volume = settings.overwrite_fuel_volume)
```
This needs to updated as well

ssankalp26 replied:
Done

ssankalp26 commented on RCAIDE/Library/Mission/Solver/converge.py:
```diff
@@ -6,6 +6,7 @@
 # ----------------------------------------------------------------------------------------------------------------------
 #  IMPORT
 # ----------------------------------------------------------------------------------------------------------------------
+from logging import raiseExceptions
```
This seems to be a unintentional add-on

ssankalp26 replied:
Done

ARmolloy commented on RCAIDE/Library/Methods/Geometry/LOPA/compute_layout_of_passenger_accommodations.py:
```diff
@@ -149,10 +158,12 @@ def create_class_seating_map_layout(cabin,cabin_class,cabin_class_origin, side_c
     if type(cabin) == RCAIDE.Library.Components.Fuselages.Cabins.Side_Cabin:
+        offset = 0
+        if cabin.side_ailse: 
+            offset = cabin_class.aile_width      
```
We have inconsistent spelling here. Doesn't matter which, but we should probably stick with one

ssankalp26 replied:
**lets make everything aisle which is the right spelling **

ARmolloy replied:
Done

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -73,55 +73,64 @@ def compute_fuselage_integral_tank_fuel_volume(fuel_tank,fuselage):
+            fuel_tank.mass_properties.center_of_gravity       = list(segment_tank_moment / total_fuel_mass) 
+            fuel_tank.origin      = [[origin_x, origin_y, origin_z]]             
+            fuel_tank.fuel.origin = [[origin_x, origin_y, origin_z]]
+        
```
Overwritting Fuel Volume here will cause issues when we have functions after this in the case of Liquid Hydrogen Tanks, and If we have made it this afar it would mean that the user would want to overwrite it. Keeping the overwrite flag in the highest level of compute_volume makes its easier

ssankalp26 replied:
Done

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -177,55 +186,63 @@ def compute_wing_integral_tank_volume(fuel_tank,wing):
-        else:
-            fuel_tank.fuel.mass_properties.mass = float(total_fuel_volume *  fuel_tank.fuel.density)  
+        
+        if overwrite_fuel_volume:            
```
Overwritting Fuel Volume here will cause issues when we have functions after this in the case of Liquid Hydrogen Tanks, and If we have made it this afar it would mean that the user would want to overwrite it. Keeping the overwrite flag in the highest level of compute_volume makes its easier

ARmolloy commented on RCAIDE/Library/Methods/Geometry/Planform/compute_segment_volume.py:
```diff
@@ -0,0 +1,61 @@
+
+def compute_segment_volume(wing, inner_segment, outer_segment,n_points=401):
+    '''
+    DOCUMENTATION
```
We should add documentation at somepoint

ssankalp26 replied:
agreed

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -177,55 +186,63 @@ def compute_wing_integral_tank_volume(fuel_tank,wing):
+            fuel_tank.volume_properties.net_volume            = total_fuel_volume
+            fuel_tank.volume_properties.gross_volume          = total_fuel_volume
+             
+            if fuel_tank.fuel.mass_properties.mass != 0:
```
If we are overwriting things on the top then this check would be always pass

ssankalp26 replied:
done

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -177,55 +186,63 @@ def compute_wing_integral_tank_volume(fuel_tank,wing):
-                total_fuel_volume    += volume
+            inner_segment_idx = i
+            outer_segment_idx = i + 1
+            
```
whitespace

ssankalp26 replied:
done

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -177,55 +186,63 @@ def compute_wing_integral_tank_volume(fuel_tank,wing):
-        fuel_tank.fuel.mass_properties.mass = float(total_fuel_volume *  fuel_tank.fuel.density)  
-        fuel_tank.fuel.volume_properties.net_volume = total_fuel_volume    
+        
+        if overwrite_fuel_volume: 
```
Overwritting Fuel Volume here will cause issues when we have functions after this in the case of Liquid Hydrogen Tanks, and If we have made it this afar it would mean that the user would want to overwrite it. Keeping the overwrite flag in the highest level of compute_volume makes its easier

ssankalp26 replied:
done

ARmolloy commented on RCAIDE/Library/Methods/Noise/Correlation_Buildup/Turbofan/turbofan_engine_noise.py:
```diff
@@ -122,7 +122,7 @@ def turbofan_engine_noise(microphone_locations, turbofan, aeroacoustic_data, seg
     Diameter_primary       = turbofan.core_nozzle.diameter
     Diameter_secondary     = turbofan.fan_nozzle.diameter
-    engine_height          = turbofan.height
+    engine_height          = turbofan.origin[0][2]
```
Is this height above the ground or the height of the engine, like diameter?

ARmolloy replied:
Origin is with respect to the nose so this isn't an accurate representaiton of height above ground. This can be addressed alter, however.

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -16,13 +16,14 @@
-
+import matplotlib.pyplot as plt
 
 # ----------------------------------------------------------------------------------------------------------------------
```
dont need matplotlib here

ARmolloy replied:
Done

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Non_Integral_Tank/compute_non_integral_tank_volume.py:
```diff
@@ -333,41 +362,42 @@ def compute_wing_non_integral_tank_volume(fuel_tank, wing):
+                inner_segment.tank_percent_span_location, tank_volume_o, tank_volume_i\
+                                    = compute_wing_non_integral_tank_fuel_volume(fuel_tank,wing,inner_segment,inner_segment_index, outer_segment,outer_segment_index, tank_percent_span_location)
+            except: 
+                fuel_tanks.pop(fuel_tank.tag)
```
add a warning that the respective tank cannot be fit into the segment space and it is being deleted
 print(f"[WARNING] Tank '{fuel_tank.tag}' does not fit in the segment. Removing from list.")

ssankalp26 replied:
Added

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -347,17 +364,17 @@ def compute_segmented_wing_integral_tank_fuel_volume(wing,inner_segment,outer_se
     inner_front_rib_length  = inner_segment_chord * (abs(inner_front_rib_yu) + abs(inner_front_rib_yl))
     inner_rear_rib_length   = inner_segment_chord * (abs(inner_rear_rib_yu) + abs(inner_rear_rib_yl) )
-    inner_wingbox_length    = inner_segment_chord * (inner_segment.fuel_tank.percent_chord_end_location -inner_segment.fuel_tank.percent_chord_start_location)  
+    inner_wingbox_length    = inner_segment_chord * (fuel_tank.segments_percent_chord_end[0] -fuel_tank.segments_percent_chord_start[0])  
```
This need not be an array or are we considering tapering tanks.

ssankalp26 replied:
Done

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -177,55 +186,63 @@ def compute_wing_integral_tank_volume(fuel_tank,wing):
-        segment_tank_moment = np.array([0.0, 0.0, 0.0])
-        seg_tags = list(wing.segments.keys()) 
+
+    if len(wing.segments) > 1:      
```
THis will not consider all the segments between the bounds if there are any,
The code to do that is here but was deleted.
  ```
 # Collect all segment tags between start and end (inclusive)
        collect = False
        seg_tags = []
        for segment in wing.segments:
            if segment.tag == seg_bounds[0]:
                collect = True
            if collect:
                seg_tags.append(segment.tag)
            if segment.tag == seg_bounds[1]:
                break
```

ssankalp26 replied:
Done

ARmolloy committed:
Aisle refactoring and cleanup

Cleanup of unneeded import from compute_non_integral_tank_volume and comment in turbofan engine noise

ssankalp26 committed:
Refactor fuel tank volume computation and settings

Values in fuel tank volume test are updates since tanks are being dropped if they dont fit and it doesnt have any volume so there is a huge reduction since three tanks dont fit
Replaces 'overwrite_fuel_volume' with 'compute_fuel_volume' and 'update_max_fuel' flags throughout geometry and fuel tank volume computation routines. Refactors segment chord bounds handling, updates method signatures, and improves logic for fuel volume calculations and error checking. Also removes unused import and cleans up related code for clarity and consistency.

ssankalp26 committed:
Merge branch 'bug_fix-aft_tank_volume_computation' of https://github.com/leadsgroup/RCAIDE_LEADS into bug_fix-aft_tank_volume_computation

ssankalp26 committed:
Update fuel_tank_volume_test.py

ssankalp26 committed:
Update generate_3d_fuel_tank_points.py

ssankalp26 committed:
Update battery_cell.py

Minimal changes possibly due to unique geometry flag

ssankalp26 committed:
Update non-integral fuel tank volume logic and tests

Fixed parameter usage in compute_wing_non_integral_tank_volume.py for correct span location and chord percent calculations. Updated fuel tank test setup in fuel_tank_volume_test.py to use new segment bounding and chord bounds attributes, corrected tank type, and adjusted expected volume for improved test accuracy.
